### PR TITLE
LF: Deprecate com.daml.lf.data.TryOps.Bracket

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala [hunk 1]
```diff
@@ -1,87 +0,0 @@
...
-package com.daml.lf.data
-
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AnyWordSpec
-import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
-import com.daml.lf.data.TryOps.Bracket.bracket
-
-import scala.util.{Failure, Success, Try}
-
-class TryOpsTest extends AnyWordSpec with Matchers with ScalaCheckDrivenPropertyChecks {
```
Comment (S11001001) [inline daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala:13]: Alternatively, you could deprecate this class.

**Review by S11001001 — APPROVED** [review]

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/TryOps.scala [hunk 2]
```diff
@@ -17,6 +17,7 @@ private[daml] object TryOps {
       b <- tb
     } yield f(a, b)
 
+  @deprecated("use scala.util.Using", since = "1.16.0")
```
Comment (stefanobaghino-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/TryOps.scala:20]: I think this is used only internally, any reason to leave it around?

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala [hunk 3]
```diff
@@ -90,7 +88,7 @@ object GenDarReader {
     }
 
     private[GenDarReader] def readDalfNames: Try[Dar[String]] =
-      bracket(get(ManifestName).map(_.toInputStream))(is => Try(is.close()))
+      get(ManifestName)
```
Comment (stefanobaghino-da) [inline daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala:91]: Does this change the guarantees given to the callers with regards to cleaning up?

File: daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala [hunk 4]
```diff
@@ -1,87 +0,0 @@
-// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (cocreature) [inline daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala:1]: if we leave it around we should imho also leave the tests around.

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala (hunk 3 above)
Comment (cocreature) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala:91]: There is no resource to cleanup. Previously, we opened the stream here and closed it here. Now `DarManifestReader.dalfNames` opens and closes the stream.

**Review by cocreature — APPROVED** [review]
Thanks!

File: daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DarReader.scala:91]: `get(ManifestName)` returns a immutable value no need for clean up here anymore.
The creation of the `InputStream` (with its cleanup is done) in `DarManifestReader.scala` line 18.

File: daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala (hunk 4 above)
Comment (remyhaemmerle-da) [reply daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala:1]: :+1:

File: daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/TryOpsTest.scala:13]: :+1:

File: ledger-service/jwt/BUILD.bazel [hunk 5]
```diff
@@ -34,24 +38,13 @@ da_scala_library(
...
     ],
     deps = [
-        "//daml-lf/data",
-        "@maven//:com_auth0_java_jwt",
-        "@maven//:com_auth0_jwks_rsa",
-        "@maven//:com_google_guava_guava",
-        "@maven//:org_slf4j_slf4j_api",
+        ":jwt",
     ],
 )
```
Comment (remyhaemmerle-da) [inline ledger-service/jwt/BUILD.bazel:49]: I am not sue why there is both the source 
```srcs = glob(["src/main/scala/**/*.scala"]),```
and the library
```
 ":jwt",
```

Comment (cocreature) [reply ledger-service/jwt/BUILD.bazel:49]: looks broken, the library should exclude the `Main` class and then the binary should depend on the lib and the source for the Main class. But your change seems fine for now.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/TryOps.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/TryOps.scala:20]: I just prefer to deprecate anything which is public.

**Pull request merged**